spec/code_block_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'spec_helper'
+require 'kramdown'
+require 'ap'
+require 'pp'
+site = File.join(File.dirname(__FILE__), '..','content', '**', '*.md')
+PAGES = Dir.glob(site)
+
+def extract_code(element,prev_element)
+  codes = [] 
+  if element.type == :codespan \
+    && !element.value.nil? \
+    && !codes.include?(element.value) \
+    && ((prev_element.children.size == 1) || (element.value[/^\n/])) #element value hasn't got any sibling or begin with '\n' 
+      codes.push element.value
+  end
+  element.children.each do |child|
+    codes += extract_code(child, element)
+  end
+  codes
+end
+
+PAGES.each do |page|
+  generated_page = Kramdown::Document.new(File.read(page))
+  codes = extract_code(generated_page.root,generated_page)
+  
+  describe 'Code blocks on markdown' do
+
+    
+    before :each do 
+      codes.each do |code|
+        code.gsub! "\n",''
+      end
+    end  
+    it 'have good syntax' do
+      codes.each do |code|
+        if code[/^\$/] #code starts with '$'
```

Comment:
It seems to me like we want to have different tests for various `code-lines` (commands).
Meaning this logic could be deciding for the test enablement, but from the level above.